src/index.tsx

```diff
@@ -115,21 +116,25 @@ const AutowidthInput: React.FC<AutowidthInputProps> = forwardRef<
                     width = placeholderWidth;
                 }
 
+                if (width < +minWidth) {
+                    width = +minWidth;
+                }
+
                 if (width) {
                     setInputWidth(width + +extraWidth);
                     if (onAutosize) onAutosize(width);
                 }
             } else if (props.placeholder && placeholderWidth) {
                 /* If no input value exists, check for placeholder value and update the sizer accordingly  */
 
-                setInputWidth(placeholderWidth + +extraWidth);
+                setInputWidth(Math.max(+minWidth, placeholderWidth) + +extraWidth);
 
                 if (onAutosize) onAutosize(placeholderWidth);
             } else if (sizerRef.current) {
                 /* If no input value or placeholder exists, update the sizer to the width of the "extraWidth" prop (default is 16) */
 
                 setInputWidth(+extraWidth);
-                if (onAutosize) onAutosize(+extraWidth);
+                if (onAutosize) onAutosize(+minWidth);
```

Comment:
This was the only place where `extraWidth` was included in `onAutosize` call, it felt odd and was inconsisted.

Reply:
Would it make more sense to add extraWidth to the other onAutosize callback instead of removing it from this one?

Reply:
I mean, one way or another. Also, it need to be consistent with `minWidth` as one might collect sizes from multiple inputs and apply same `minWidth={Math.max(...inputSizes)}` to all of them